app/controllers/collections_controller.rb

```diff
@@ -0,0 +1,227 @@
+class CollectionsController < ApplicationController
+  load_and_authorize_resource
+  def index
+    @collections = Collection.all
+    @public_collections = Collection.where(is_private: false) 
+    @public_collections = @public_collections.paginate(page: params[:public_page], :per_page => 10)
+
+    if !current_user.nil?
+      @owned_collections = current_user.owned_collections
+      @private_collections = @owned_collections.where(is_private: true)
+      @private_collections = @private_collections.paginate(page: params[:private_page], :per_page => 10)
+    end
+    render :layout => "custom"
+  end
+
+  def new
+    if !current_user.nil?
+      @collection = Collection.new
+      render :layout => "custom"
+    else
+      redirect_to({ action: "index" }, :flash => { :warning => "You must be logged in to create a collection." })
+    end
+  end
+
+  def create
+    @collection = Collection.new(params[:collection])
+    url = request.referrer
+    if url =~ /\/([^\/]+)(?=\/[^\/]+\/?\Z)/
```

Comment:
You should be able to create a method with child methods and cases without scraping method and case ids from the url. I think you are already doing it this way in the `update` method?